### PR TITLE
update:add validation functions to validate user input at init.ts script

### PR DESCRIPTION
**Pull request by sinyo-matu**
## Background

In init.ts,
I think user input may need some validation to reduce unexpected or confusing behavior.

Move init.ts script contents into src/ may make our code cleaner
 
## What Changed
1. Adding two validation functions: `isValidaUrl` and `isValidDir`
2. Moving init.ts script contents input src/intiFile.ts

### CODE REVIEW

File: src/initFile.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  try {
+    new URL(input);
+  } catch (_) {
+    return false;
+  }
+  return true;
+}
+
+function validateUrl(inputUrl: string | null): string {
+  if (!inputUrl || !isUrl(inputUrl.trim())) return "http://example.com";
```
Comment (xstevenyung) [inline src/initFile.ts:13]: I think I would rather throw an error here or even better display an error message and ask again for a prompt

File: src/initFile.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  return inputUrl;
+}
+
+function validateDir(inputDir: string | null): string {
+  if (!inputDir || inputDir.trim().length === 0) return "./static";
```
Comment (xstevenyung) [inline src/initFile.ts:18]: Same here as comment above

**Review by xstevenyung — CHANGES_REQUESTED** [review]

File: src/initFile.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  }
+
+  let staticPath = prompt(
+    "Please input your site's static folder path (skip by enter):",
+    "./static",
+  );
+
+  if (!staticPath || !isValidDir(staticPath)) {
+    console.log("Invalid folder input! Setting to default value:'./static'");
+    staticPath = ".static";
```
Comment (xstevenyung) [inline src/initFile.ts:83]:
```suggestion
    staticPath = "./static";
```

**Review by xstevenyung — APPROVED** [review]
Yeah I like this approach a lot, just some minor changes and we should be good to go

**Review by xstevenyung — CHANGES_REQUESTED** [review]
Oops wrong button

File: src/initFile.ts (hunk 3 above)
Comment (sinyo-matu) [reply src/initFile.ts:83]: Thank you for pointing them out!

**Comment by sinyo-matu** [comment]
@xstevenyung 
If you got time, please have a check!

**Review by xstevenyung — APPROVED** [review]

**Comment by xstevenyung** [comment]
thanks a lot @sinyo-matu , amazing work ! 👍

**Pull request merged**